apps/snitch_core/lib/core/data/model/tax_category.ex

```diff
@@ -151,15 +151,30 @@ defmodule Snitch.Data.Model.TaxCategory do
 
   Takes as input the `instance` of the TaxCategory to be deleted.
   """
-  @spec delete(TaxCategory.t()) ::
+  @spec delete(TaxCategory.t() | integer) ::
           {:ok, TaxCategory.t()}
           | {:error, Ecto.Changeset.t()}
+  def delete(id) when is_integer(id) do
+    params = %{deleted_at: DateTime.utc_now(), id: id}
+
+    case QH.update(TaxCategory, params, Repo) do
+      {:ok, changeset} ->
+        {:ok, changeset}
+
+      {:error, changeset} ->
+        {:error, changeset}
+    end
+  end
+
   def delete(instance) do
     params = %{deleted_at: DateTime.utc_now()}
 
     case QH.update(TaxCategory, params, instance, Repo) do
```

Comment:
Can the output of `QH.update(TaxCategory, params, instance, Repo)` returned directly?